pkg/controller/certificatemanager/certificatemanager.go

```diff
@@ -284,22 +360,23 @@ func (cm *certificateManager) getKeyPair(cli client.Client, secretName, secretNa
 		return nil, nil, err
 	}
 
-	timeInvalid := x509Cert.NotAfter.Before(time.Now()) || x509Cert.NotBefore.After(time.Now())
 	// Get specific usages to check for certs that are utilized for mTLS with Linseed
 	requiredKeyUsages := certkeyusage.GetCertKeyUsage(secretName)
 	invalidKeyUsage := !HasRequiredKeyUsage(x509Cert, requiredKeyUsages)
+	timeInvalid := x509Cert.NotAfter.Before(time.Now()) || x509Cert.NotBefore.After(time.Now())
 	if timeInvalid || invalidKeyUsage {
 		if !readCertOnly && strings.HasPrefix(x509Cert.Issuer.CommonName, rmeta.TigeraOperatorCAIssuerPrefix) {
 			if cm.keyPair.CertificateManagement != nil {
 				// When certificate management is enabled, we can simply return a certificate management key pair;
 				// the old secret will be deleted automatically.
-				return certificateManagementKeyPair(cm, secretName, nil), nil, nil
+				return certificateManagementKeyPair(cm, secretName, secretNamespace, dnsNames), nil, nil
 			}
 
 			if invalidKeyUsage {
 				log.Info("secret %s/%s must specify ext key usages: %+v", secretNamespace, secretName, requiredKeyUsages)
 			}
 			// We return nil, so a new secret will be created for expired (legacy) operator signed secrets.
+			cm.log.V(2).Info("KeyPair is an expired legacy operator cert, make a new one", "name", secretName)
```

Comment:
How about bumping the log level to warning? It may actually be of great value in some edge cases.